libi2pd/util.cpp

```diff
@@ -143,8 +146,15 @@ namespace util
 	}
 
 	void SetThreadName (const char *name) {
-#if defined(__APPLE__) && !defined(__powerpc__)
+#if defined(__APPLE__)
+# if (!defined(MAC_OS_X_VERSION_10_6) || \
```

Comment:
`not 10.6`??? It will stop checking here leading to part where is nothing is done.

Reply:
@r4sas `MAC_OS_X_VERSION_10_6` _is_ defined for all macOS starting from 10.6. There is no forward definition. As you can see, on 10.5 it is _not_ defined: https://github.com/phracker/MacOSX-SDKs/blob/master/MacOSX10.5.sdk/usr/include/AvailabilityMacros.h

So the logic reads: if not defined `MAC_OS_X_VERSION_10_6` = all macOS prior to 10.6 (Intel has to be included, so having only `__POWERPC__` is insufficient).